corehq/apps/app_manager/views/download.py

```diff
@@ -249,6 +249,10 @@ def resolve_path(path):
         if request.app.copy_of:
             if request.META.get('HTTP_USER_AGENT') == 'bitlybot':
                 raise Http404()
+            elif path in ('CommCare.jad', 'CommCare.jar'):
+                request.app.create_jadjar(save=True)
```

Comment:
After the first request, this won't be needed anymore because the documents will be found.  I added https://github.com/dimagi/commcare-hq/commit/220224673aea06d25c4f91e8ce56d4955c2819a4 to deal with the race condition mentioned in https://github.com/dimagi/commcare-hq/issues/8422